statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -106,23 +114,46 @@ def hannan_rissanen(endog, ar_order=0, ma_order=0, demean=True,
     # Create a spec, just to validate the initial autoregressive order
     _ = SARIMAXSpecification(endog, ar_order=initial_ar_order)
 
+    # Unpack fixed and free ar/ma lags, ix, and params (fixed only)
+    params_info = _package_fixed_and_free_params_info(
+        fixed_params, spec.ar_lags, spec.ma_lags
+    )
+
     # Compute lagged endog
-    # (`ar_ix`, and `ma_ix` below, are to account for non-consecutive lags;
-    # for indexing purposes, must have dtype int)
-    ar_ix = np.array(spec.ar_lags, dtype=int) - 1
-    lagged_endog = lagmat(endog, max_ar_order, trim='both')[:, ar_ix]
+    lagged_endog = lagmat(endog, max_ar_order, trim='both')
```

Comment:
Delaying the column slicing here for easier separation between cols with free params vs. cols with fixed params later.